src/AspNet.Security.OAuth.Zoho/ZohoAuthenticationHandler.cs

```diff
@@ -51,6 +51,30 @@ public partial class ZohoAuthenticationHandler : OAuthHandler<ZohoAuthentication
         return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
     }
 
+    protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync()
+    {
+        var location = Context.Request.Query["location"];
+
+        using var requestMessage = new HttpRequestMessage(HttpMethod.Get, ZohoAuthenticationDefaults.ServerInfoEndpoint);
+        requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+        requestMessage.Version = Backchannel.DefaultRequestVersion;
+
+        using var response = await Backchannel.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.ServerInfoErrorAsync(Logger, response, Context.RequestAborted);
+            throw new HttpRequestException("An error occurred while retrieving the server info.");
+        }
+
+        using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        var domain = payload.RootElement.GetProperty("locations").GetProperty(location.ToString());
+        Options.UserInformationEndpoint = $"{domain}{ZohoAuthenticationDefaults.UserInformationPath}";
```

Comment:
You can't do that: the `Options` object is shared globally and `HandleRemoteAuthenticateAsync()` might be called concurrently (e.g if multiple Zoho authentication are handled concurrently).

(feel free to hardcode the endpoints to avoid an extra query: they are unlikely to change anyway)